src/plugins/intel_cpu/src/infer_request.h

```diff
@@ -4,106 +4,94 @@
-                     const std::vector<std::shared_ptr<const ov::Node>>& outputs,
-                     std::shared_ptr<ExecNetwork> execNetwork_)
-    : IInferRequestInternal(inputs, outputs), execNetwork(execNetwork_) {}
+    void create_infer_request();
+    InferenceEngine::Precision norm_to_input_supported_prec(const std::pair<const std::string, ov::Tensor>& input) const;
+    void pushInput(const std::string& inputName, ov::Tensor& inputBlob, InferenceEngine::Precision dataType);
 
-    void CreateInferRequest();
-    InferenceEngine::Precision normToInputSupportedPrec(const std::pair<const std::string, InferenceEngine::Blob::Ptr>& input) const;
-    void pushInput(const std::string& inputName, InferenceEngine::Blob::Ptr& inputBlob, InferenceEngine::Precision dataType);
-
-    virtual void initBlobs() = 0;
-    virtual void PushInputData() = 0;
+    void init_tensor(const std::string& name);
+    void push_input_data();
 
     Graph* graph = nullptr;
-    std::unordered_map<std::string, void*> externalPtr;
+    mutable std::unordered_map<std::string, void*> external_ptr;
 
 private:
-    void PushStates();
-    void PullStates();
-    void redefineMemoryForInputNodes();
-
-    std::shared_ptr<ExecNetwork>        execNetwork;
-    openvino::itt::handle_t             profilingTask;
-    std::vector<std::shared_ptr<InferenceEngine::IVariableStateInternal>> memoryStates;
-    AsyncInferRequest*                  _asyncRequest = nullptr;
+    void push_states();
+    void pull_states();
+    void redefine_memory_for_input_nodes();
+
+    void update_external_inputs();
+    InferenceEngine::TensorDesc create_tensor_desc(const ov::Tensor& tensor);
+
+    // Transformation shouldn't change model's input/output's precision, but actually it does.
+    // Some additional methods will handle it.
+    const ov::Output<const ov::Node>& get_internal_port(const ov::Output<const ov::Node>& port) const;
+    ov::Tensor get_port_tensor(const ov::Output<const ov::Node>& port) const;
+    // Check this port is original model port or compiled model's port, return true for compiled model's port
+    // This is because if precision has been changed in compiled model, it need distinguish them
+    bool check_compiled_model_port(const ov::Output<const ov::Node>& port) const;
+    mutable std::unordered_map<std::string, ov::Output<const ov::Node>> m_orig_ports_map;
+    // Store external tensor due to precision changes
+    mutable std::unordered_map<std::string, ov::Tensor> m_aux_tensors;
+    mutable std::unordered_map<std::string, bool> m_port_precision_changed;
+    bool m_is_legacy_api = false;
+
+    std::shared_ptr<const CompiledModel> m_compiled_model;
+    openvino::itt::handle_t m_profiling_task;
+    std::vector<std::shared_ptr<ov::IVariableState>> m_memory_states;
+    AsyncInferRequest* m_asyncRequest = nullptr;
+
+    mutable std::unordered_map<std::string, ov::Output<const ov::Node>> m_input_ports_map;
```

Comment:
Is it possible to use `std::shared_ptr<Output>` instead of `Output` copy here?
This point is applicable to all maps here.

Reply:
It is a good point!
but I still think maybe` ov::Output<const ov::Node>` is better than `std::share_ptr<ov::Output<const ov::Node>>`, because:
1) Plugin API adopt ` ov::Output<const ov::Node>`  as input arguments of function, such as:
     https://github.com/openvinotoolkit/openvino/blob/b69190c2c164a460ecbbf74615badc6febbda791/src/inference/dev_api/openvino/runtime/iinfer_request.hpp#L54-L87

2). ov::Output<const ov::Node> only contains a node ptr and index, std::shared_ptr doesn't have too much benefit:
https://github.com/openvinotoolkit/openvino/blob/b69190c2c164a460ecbbf74615badc6febbda791/src/core/include/openvino/core/node_output.hpp#L110-L113

Reply:
Ok